app/scripts/metamask-controller.js

```diff
@@ -2000,6 +2000,11 @@ export default class MetamaskController extends EventEmitter {
         ],
       }),
       trace,
+      decodingApiUrl: process.env.DECODING_API_URL,
+      isDecodeSignatureRequestEnabled: () =>
+        this.preferencesController.state.useExternalServices === true &&
+        this.preferencesController.state.useTransactionSimulations &&
```

Comment:
Is this relevant here since this isn't related to transactions?

Reply:
we use it on signature pages for simulation section and thus plan is to use it here also, we may add another flag for this in future

Reply:
Do we actually rely on the preference in the UI though, I thought this was only used by the transaction controller to query the simulation API?

Reply:
Yep we do https://github.com/MetaMask/metamask-extension/blob/68ba6351f710a3a2a18332b075cd6b999baf09dd/ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign.tsx#L58